src/vm/eventpipebuffermanager.h

```diff
@@ -163,6 +174,30 @@ class EventPipeBufferList
 #endif // _DEBUG
 };
 
+EXTERN_C inline EventPipeBufferList* STDCALL GetThreadEventBufferList();
+EXTERN_C inline void STDCALL SetThreadEventBufferList(EventPipeBufferList* bl);
+
+
+// This struct is a TLS wrapper around a pointer to thread-specific EventPipeBufferList
+// The struct wrapper is present mainly because we need a way to free the EventPipeBufferList
+// when the thread that owns it dies. Placing this struct as a TLS variable will call ~ThreadEventBufferList()
+// when the thread dies so we can free EventPipeBufferList in the destructor.  
+struct ThreadEventBufferList
+{
+    Volatile<EventPipeBufferList*> m_pThreadEventBufferList;
```

Comment:
Does this need to be volatile? As far as I can tell, this is only ever accessed from the current thread. It means that it does not need to be volatile.

Reply:
I marked it as volatile because it was volatile before. But yeah it's only accessed from the current thread so I'll remove it. Thanks.